### DISCUSSION TITLE
Idea for targets / targetopia package

### DISCUSSION THREAD

**Discussion by shah-in-boots** [ropensci/targets#558] [Ideas]
Hello @wlandau,

I'm a huge fan of {targets}. I do a epidemiology-based research and signal processing, and in each of my research projects, I set it up very similarly with targets. These are not necessarily very "complex" projects, but very causal in their nature.

A pattern I've seen is that a lot of analyses are very similar per project/paper (e.g. a table 1 overview, regressions, splines, etc). What I wrote for myself is R code that helps to set data, add specific hypothesis testing, and then collecting the results. A huge part of this is the type of hypothesis (e.g. sequential adjustment), in setting of epidemiology and causal modeling. 

As I was going through your work, I was wondering if there would be utility in a {targets} extension package that helps to encapsulate that workflow?

The other packages I've seen, such as {stantargets} are more so wrapper packages to help simplify the interface in other projects. 

I wanted to get a general thought from your end to see if this makes sense.

For example...

```r
# _targets.R
library(targets)
library(target_package)

list(
  tar_correlations(
    name = "correlations",
    vars = any_of(x1, x2, x3)
  ),
  tar_distributions(
    name = "normal_dist",
    vars = c(y1, y2, x1, x2),
    visualize = TRUE,
  ),
  tar_hypothesis(
    name = "linear",
    exposure = x1,
    outcomes = y1,
    covariates = c(x2, x3),
    test = linear_reg() %>% set_engine("lm") # such as from parsnip
  ),
)
```

**Comment by wlandau** [comment]
Your proposal feels almost like a domain-specific language for hypothesis testing in epidemiology workflows, which seems appealing. I do not personally plan to write such an extension myself, but this sort of thing is possible using the recommendations at https://wlandau.github.io/targetopia/contributing.html. As you mentioned, the goal would be to abstract away the fine details of structuring the workflow in order to free up users to think about the domain knowledge and the high-level goals of the analysis.

Looking at what you have, it seems like `tar_hypothesis()` would go through a hypothesis testing workflow (how is the dataset specified?). I wonder if there is some shorthand for `test` so that users do not have to supply arbitrary custom code. `tar_correlations()` and  `tar_distributions()` look like they add convenience if these are common exploratory analyses on the data.

**Reply by shah-in-boots** [reply]
I briefly wrote those examples as a trivial example. I'll go through the documentation you provide and see if I can write something that may prove to be helpful/practical. I've just found that every one of my research projects starts with a `_targets.R` file, and I find myself copy/pasting code over and over.
